mypy/build.py

```diff
@@ -416,23 +416,12 @@ def parse_file(self, id: str, path: str, source: str) -> MypyFile:
         tree = parse(source, path, self.errors, options=self.options)
         tree._fullname = id
 
-        # We don't want to warn about 'type: ignore' comments on
-        # imports, but we're about to modify tree.imports, so grab
-        # these first.
-        import_lines = set(node.line for node in tree.imports)
-
-        # Skip imports that have been ignored (so that we can ignore a C extension module without
-        # stub, for example), except for 'from x import *', because we wouldn't be able to
-        # determine which names should be defined unless we process the module. We can still
-        # ignore errors such as redefinitions when using the latter form.
-        imports = [node for node in tree.imports
-                   if node.line not in tree.ignored_lines or isinstance(node, ImportAll)]
-        tree.imports = imports
-
         if self.errors.num_messages() != num_errs:
             self.log("Bailing due to parse errors")
             self.errors.raise_error()
 
+        # We don't want to warn about 'type: ignore' comments on imports
```

Comment:
I don't think this is the right decision.  You're not intended to always run with `--warn-unused-ignores` in CI, for example, because this _can_ be a problem in other places (like when you're transitioning between mypy releases).  `--warn-unused-ignores` is intended to be run periodically to clean up unnecessary `# type: ignore` comments and is intended to require human judgement in the loop.  If something is even just potentially unused, we should warn about it here.

Put another way: if we don't warn about type ignore comments on imports, they'll never be removed.